Fabric.Identity.UnitTests/Services/DecryptionServiceTests.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Security.Cryptography;
+using System.Text;
+using EnsureThat;
+using Fabric.Identity.API.Configuration;
+using Fabric.Identity.API.Services;
+using Xunit;
+using Moq;
+
+namespace Fabric.Identity.UnitTests.Services
+{
+    public class DecryptionServiceTests
+    {
+        [Fact]
+        public void DecryptionService_ThrowsArgumentNullException_NullCertificateService()
+        {
+            Assert.Throws<ArgumentNullException>(() => new DecryptionService(null));
+        }
+        [Fact]
+        public void DecryptString_DecryptsEncryptedString()
+        {
+            // Arrange
+            var privateKey = GetPrivateKey();
+            var stringToEncrypt = Guid.NewGuid().ToString();
+            var encryptedString = $"{DecryptionService.EncryptionPrefix}{EncryptString(privateKey, stringToEncrypt)}";
+
+            var mockCertificateService = GetMockCertificateService(privateKey);
+            var signingCertificateSettings = new SigningCertificateSettings();
+            var decryptionService = new DecryptionService(mockCertificateService);
+
+            // Act
+            var decryptedString = decryptionService.DecryptString(encryptedString, signingCertificateSettings);
+
+            // Assert
+            Assert.Equal(stringToEncrypt, decryptedString);
+        }
+
+        [Fact]
+        public void DecryptString_ReturnsNonEncryptedString()
+        {
+            // Arrange
+            var privateKey = GetPrivateKey();
+            var clearTextstringToDecrypt = Guid.NewGuid().ToString();
+
+            var mockCertificateService = GetMockCertificateService(privateKey);
+            var signingCertificateSettings = new SigningCertificateSettings();
+            var decryptionService = new DecryptionService(mockCertificateService);
+
+            // Act
+            var decryptedString = decryptionService.DecryptString(clearTextstringToDecrypt, signingCertificateSettings);
+
+            // Assert
+            Assert.Equal(clearTextstringToDecrypt, decryptedString);
+        }
+
+        [Theory]
+        [InlineData(null)]
```

Comment:
I like this part about XUnit vs MStest.  These InlineData allow you to compress 3 tests into one.  Cuts out a bunch of bloat.